### PR TITLE
Fix behaviour of LIMIT 0

### PR DESCRIPTION
**Pull request by systay**
Using LIMIT to batch updates is a common pattern. LIMIT 0 should mean that nothing is changed in the graph, but this was not the case before this change.

changelog: Using `LIMIT` with zero or a negative argument will never pull any rows.

### CODE REVIEW

File: community/cypher/cypher-compiler-3.2/src/main/scala/org/neo4j/cypher/internal/compiler/v3_2/pipes/QueryState.scala [hunk 1]
```diff
@@ -45,6 +45,8 @@ class QueryState(val query: QueryContext,
                    new SingleThreadedLRUCache(maxSize = 16)) {
   private var _pathValueBuilder: PathValueBuilder = null
 
+  def createEmptyExecutionContext(): ExecutionContext = initialContext.getOrElse(ExecutionContext.empty)
```
Comment (Mats-SX) [inline community/cypher/cypher-compiler-3.2/src/main/scala/org/neo4j/cypher/internal/compiler/v3_2/pipes/QueryState.scala]: This name seems a bit misleading. Perhaps `createOrGetInitialContext()`?

File: community/cypher/cypher-compiler-3.2/src/main/scala/org/neo4j/cypher/internal/compiler/v3_2/pipes/SkipPipe.scala [hunk 2]
```diff
@@ -34,27 +34,8 @@ case class SkipPipe(source: Pipe, exp: Expression)
     if(input.isEmpty)
       return Iterator.empty
 
-    implicit val s = state
+    val skip = asInt(exp(state.createEmptyExecutionContext())(state))
```
Comment (Mats-SX) [inline community/cypher/cypher-compiler-3.2/src/main/scala/org/neo4j/cypher/internal/compiler/v3_2/pipes/SkipPipe.scala]: Nice simplification!

**Comment by systay** [comment]
Your kind feedback has been acted on, @Mats-SX

**Comment by Mats-SX** [comment]
@systay I'm not convinced this is where we will end up with `LIMIT 0` on updating queries, but this does fix the weird pull-one behaviour of read-only queries. I'll merge it.

**Pull request merged**